src/ui/public/directives/paginate.js

```diff
@@ -40,6 +40,12 @@ uiModules.get('kibana')
     },
     controllerAs: 'paginate',
     controller: function ($scope, $document) {
+      // Unfortunately this controller relies on inherited scope. Lets not fail if onPageChanged happens to not be
+      // implemented.
+      if (!$scope.onPageChanged) {
```

Comment:
I think that if it's set on the child, it'll just 'hide' the parent's declaration:
>When AngularJS evaluates {{name}}, it first looks at the scope associated with the given element for the name property. If no such property is found, it searches the parent scope and so on until the root scope is reached. In JavaScript this behavior is known as prototypical inheritance, and child scopes prototypically inherit from their parents.